### PR TITLE
Minor aggregator API cleanup

### PR DESCRIPTION
**Pull request by inahga**
- Deletion should return 204 on not found, not 404.

    Supports https://github.com/divviup/janus/issues/1691.

- Provide explicit return type in routes

    From conversation in https://github.com/divviup/janus/pull/1686#discussion_r1290401987.

- Remove needless Halt

    I'm not sure why this is here--no other routes use Halt in this manner (should they)?

### CODE REVIEW

File: aggregator_api/src/routes.rs [hunk 1]
```diff
@@ -281,15 +278,15 @@ pub(super) async fn get_task<C: Clock>(
...
+) -> Result<Status, Status> {
     let task_id = conn.task_id_param()?;
 
     ds.run_tx_with_name("delete_task", |tx| {
         Box::pin(async move { tx.delete_task(&task_id).await })
     })
     .await
     .map_err(|err| match err {
-        datastore::Error::MutationTargetNotFound => Status::NotFound,
+        datastore::Error::MutationTargetNotFound => Status::NoContent,
```
Comment (tgeoghegan) [inline aggregator_api/src/routes.rs:289]: Were there not tests that had to be updated to account for this? If not, we should take the opportunity to add tests that exercise this.

File: aggregator_api/src/routes.rs [hunk 2]
```diff
@@ -76,19 +76,16 @@ pub(super) async fn get_task_ids<C: Clock>(
         })?;
     let pagination_token = task_ids.last().cloned();
 
-    Ok((
-        Json(GetTaskIdsResp {
-            task_ids,
-            pagination_token,
-        }),
-        Halt,
```
Comment (tgeoghegan) [inline aggregator_api/src/routes.rs:84]: Hmm, I think maybe we had to do this in order to stop other routes from also matching on the request, but I don't think any other handler would have matches on `/task_ids` so I suspect this wasn't needed.

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

File: aggregator_api/src/routes.rs (hunk 1 above)
Comment (inahga) [reply aggregator_api/src/routes.rs:289]: Whoops, that got lost in a `git stash`.

File: aggregator_api/src/routes.rs (hunk 2 above)
Comment (inahga) [reply aggregator_api/src/routes.rs:84]: Looking into it more I think it'll stop any additional handlers from running, but we don't have any at the moment. https://github.com/divviup/janus/blob/8c2761d4c438bf488227a90d66c2682b8bbb2d7b/aggregator_api/src/lib.rs#L71

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**